pkg/controller/direct/bigqueryconnection/connection_controller.go

```diff
@@ -253,6 +254,14 @@ func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperati
 	return setStatus(u, status)
 }
 
+// aws.accessRole.Identity is a output-only field in CREATE, it is required in the UPDATE.
+func merge3(desired *pb.Connection, previouslyApplied *krm.BigQueryConnectionConnectionObservedState) *pb.Connection {
```

Comment:
The function name seems confusing. Its not (currently) a 3 way merge. I would also expect something named  3 way merge to be a generic 3 way merge. This seems like a awsIdentityStateMerge() function. Can we rename it?